### PR TITLE
[Flink][KafkaIO] Finalize checkpoints during stop/drain scenario

### PR DESCRIPTION
**Pull request by kkdoon**
In the case of flink runner, when pipeline is stopped/drained, it never commits the final consumer offset to Kafka. This results in duplicate data processing as some events are re-consumed upon pipeline restart.

See #26884 for detailed explanation.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kkdoon** [comment]
Run Java_Kafka_IO_Direct PreCommit

**Comment by kkdoon** [comment]
Run Website_Stage_GCS PreCommit

**Comment by kkdoon** [comment]
Hi,

@mxm @JingsongLi  Could you kindly review the flink runner related changes (or tag someone else who might have more context).

@rangadi Could you review the kafka related change.

Thanks!

**Comment by Abacn** [comment]
Thanks for contribution! We got a couple of flink runner PRs I was trying to looking for reviewers: https://github.com/apache/beam/pulls?q=is%3Apr+is%3Aopen+label%3Aflink They look pretty valuable to me, e.g. #26193 , but I do not have much experience on flink runner. If interested would you minding helping review it, thanks again

**Comment by kkdoon** [comment]
@Abacn yeah i can take a look at that PR and hopefully provide some meaningful comments. In fact, we wanted this feature for our use-case as well so great to see it being implemented!

**Review by bvolpato — APPROVED** [review]

File: website/www/site/layouts/shortcodes/flink_java_pipeline_options.html [hunk 1]
```diff
@@ -16,7 +16,6 @@
 Use generatePipelineOptionsTableJava and generatePipelineOptionsTablePython respectively.
 Should be called before running the tests.
 -->
-<div class="table-container-wrapper">
```
Comment (Abacn) [inline website/www/site/layouts/shortcodes/flink_java_pipeline_options.html:19]: Would you mind fixing these unrelated diff? I also saw this weird diff before, this generated html changes depends on the platform.

**Comment by Abacn** [comment]
Run Java KafkaIO Performance Test
